manticore/utils/mlogging.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import sys
+import types
+
+class ContextFilter(logging.Filter):
+    '''
+    This is a filter which injects contextual information into the log.
+    '''
+    def filter(self, record):
+        if hasattr(self, 'stateid') and isinstance(self.stateid, int):
+            record.stateid = '[%d]' % self.stateid
+        else:
+            record.stateid = ''
+        return True
+
+loggers = ['MANTICORE',
+           'VISITOR',
+           'EXECUTOR',
+           'CPU',
+           'REGISTERS',
+           'SMT',
+           'MEMORY',
+           'PLATFORM']
+
+ctxfilter = ContextFilter()
+logfmt = ("%(asctime)s: [%(process)d]%(stateid)s %(name)s:%(levelname)s:"
+          " %(message)s")
+logging.basicConfig(format=logfmt, stream=sys.stdout, level=logging.ERROR)
+
+def loggerSetState(logger, stateid):
+    logger.filters[0].stateid = stateid
+
+for l in loggers:
+    logging.getLogger(l).addFilter(ctxfilter)
+    logging.getLogger(l).setState = types.MethodType(loggerSetState,
+                                                     logging.getLogger(l))
+
+def set_verbosity(setting):
```

Comment:
this should also be exposed to the api via `from manticore import set_verbosity`, which requires modification of manticore's `__init__.py`